### ISSUE TITLE
some more corrections

### ISSUE THREAD

**Issue by mathriddle** [ULeth-Math-CS/uleth-math-webwork-library#4]
I found a few more items that needed fixing. I did this in the repository I forked, so I will make a pull request. I'm a github novice, so I hope this works. But here are the files affected.

calc01006 - In line 72, the x should be evaluated at ax0001 (the x is missing in the variable name)
calc02336 - line 122, the derivative is equal to $ans3, not 1
calc02336 and calc02338 - added a missing closing parenthesis in the point being used

calc02440, 02416, and 05116. More substantial issue here. These are using trig functions and in the statement of the problem and in the solutions, the trig functions are trying to be presented without parentheses around the argument, i.e. sin x rather than sin(x). But this doesn't work inside a Compute or Formula command when generating a math object, leading to error messages about a function having no argument. So I removed the Compute/Formula commands and kept those functions just listed as (tex) strings. The places where functions were constructed that did need to be math objects for checking student answers were ok because parentheses *were* used around the arguments in those instances. I hope that makes sense.

**Comment by sean-fitzpatrick** [comment]
Thanks for this. I'm not able to reproduce errors in calc02440, 02416, and 05116, but perhaps you did more careful testing across a greater number of seeds.
In 02440 and 05116, what is the rationale for getting rid of the Math Objects code (e.g. hard-coding the answers rather than letting WeBWorK compute the derivatives)? I've tried these problems a few times and haven't been able to generate an error message.

**Comment by mathriddle** [comment]
I test the code by going to the Library Browser and viewing a problem set. I would occasionally see "there was an error rendering this problem" like in the following image for calc02440:

<img width="531" alt="image" src="https://user-images.githubusercontent.com/46355923/172689810-1877d54f-280b-4a09-863a-5ce0017c5e09.png">

I would then click the edit icon to investigate further. In the editor I could make a revision, then check it by clicking on "take action". What seems very odd is that sometimes a problem would render ok in the Library Browser, but would always fail when rendered in the editor. That may be why you were not able to reproduce the errors. 

When running calc02440 in the editor, I get a error message:
1. ERROR caught by Translator while processing problem file:tmpEdit/APEXCalculus/02-Derivatives/4-ProdQuotRules/calc02440-tmp.pg.admin.tmp
****************
ERRORS from evaluating PG file: 
 Function 'sec' is missing its input(s); see position 10 of formula at line 100 of [PG]/macros/Parser.pl
   Died within main::Formula called at line 100 of [PG]/macros/Parser.pl
   from within main::Compute called at line 55 of (eval 3392)

In calc02416, you must set $r = 4 to force the 4th part of the if block to be used. When the problem is run the following message is generated when working in the editor:

1. ERROR caught by Translator while processing problem file:tmpEdit/APEXCalculus/02-Derivatives/4-ProdQuotRules/calc02416.pg.admin.tmp
****************
ERRORS from evaluating PG file: 
 Function 'csc' is missing its input(s); see position 2 of formula at line 71 of (eval 3860)
   Died within main::Formula called at line 71 of (eval 3860)

This part of the if block contains the code:
} else {
  $g = Formula("cot t + $b");
  $gp = Formula("-csc^2 t");
The other three sections of the if block all use parentheses around the trig argument and don't generate this error message when $r is equal to 1, 2, or 3. 

I'm also getting an error message in the Library Browser with calc05116

<img width="531" alt="image" src="https://user-images.githubusercontent.com/46355923/172691422-43668f6d-6b7f-4d0b-8f21-ca6450bbd048.png">

and the message when running the problem in the editor:
1. ERROR caught by Translator while processing problem file:tmpEdit/APEXCalculus/05-Integrals/1-Antiderivatives/calc05116.pg.admin.tmp
****************
ERRORS from evaluating PG file: 
 Function 'sec' is missing its input(s); see position 4 of formula at line 55 of (eval 2087)
   Died within main::Formula called at line 55 of (eval 2087)

Line 55 has $integrandb=Formula("$b sec^2 s"); The issue here may be the squaring of sec before the variable s. I just now tried deleting the "^2" and the problem rendered fine.

I'm certainly no expert on WeBWorK, perl, and PGML syntax. I don't know exactly under what conditions these error messages are being generated. But I found that in most cases I could just replace the MathObject construction that used Formula with just a string when the function being defined was only being used in the statement of the problem or in the solutions, not as part of the answer checker. When I did this the error messages went away in both the Library Browser and the Editor. Maybe it wasn't really necessary, but it was the best I could come up with to avoid getting the error messages.

This might be something worthwhile to ask about in the WeBWorK Forum. I may do that to try to clarify when Formula("sin x") can be used versus when Formula("sin(x)") must be used.

I hope all this makes some sense!

**Comment by mathriddle** [comment]
One other comment. In 02440, looks like I just used parentheses around all the trig arguments inside the use of the Formula command to try to eliminate the error messages I was getting.

**Comment by mathriddle** [comment]
I think I found the problem upon further testing. I looks like all the problems in which I got the error messages involved attempts to use something like
Formula("sec^2 x")
or some other trig function. But it is the "^2" that appears to be the issue. Formula("sec x") works fine. 

So maybe the only change that needs to be made is to replace things like Formula("sec^2 x") with Formula("sec(x)^2")?

**Comment by mathriddle** [comment]
Further testing shows that Formula("sec^2 x") produces an error but Formula("sec^2x") works correctly.

**Comment by sean-fitzpatrick** [comment]
Rather than designing the pull request: can you make these edits and push another commit?
You can continue to add new commits to a pull request until it is merged.
And I think it's easier for you to make the change than me.

Thanks for your careful testing of these problems! They haven't had a lot of attention in the last two years, aside from fixing things when we get an error report from a student.

**Comment by mathriddle** [comment]
Okay, I made the changes in my copy and committed them. In 02416 I kept my
original change which was to add parentheses around the argument in the
trig function. This makes it consistent with the other three parts of the
if statement where parentheses were used. But in the other two files I just
deleted the space between the ^2 and the variable. Also, in 02440 I had
previously added the derivative rule for csc(x) to the solution since the
solution only mentioned sec(x) but the problem could be about either sec(x)
or csc(x). I kept that in my revision.

So if I understand the process correctly, that's all I need to do and the
changes should show up in the pull request I previously made. Let me know
if I need to do anything else.

Larry
